Kong/docs.konghq.com
# docs(kic): add upstream TLS verification guide

### Description

Adds a guide covering upstream TLS verification configuration features added in KIC 3.4. 

Part of https://github.com/Kong/docs.konghq.com/issues/8133 (this guide will have to be extended with Gateway API instructions once https://github.com/Kong/kubernetes-ingress-controller/issues/6631 is implemented).

### Testing instructions

KIC nightly image has to be used for testing:
```shell
helm upgrade --install kong kong/ingress -n kong --create-namespace --set controller.ingressController.image.repository=kong/nightly-ingress-controller --set controller.ingressController.image.tag=2024-11-28 --set controller.ingressController.image.effectiveSemver=3.4.0

```

Preview link: https://deploy-preview-8201--kongdocs.netlify.app/kubernetes-ingress-controller/unreleased/guides/security/verify-upstream-tls/

### Checklist
